enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,240 @@
+	// DeviceDiscoverPolicies are the list of policies that AutoDetectVolume's controller
+	// will be looking for during detection
+	DeviceDiscoveryPolicies DeviceDiscoveryPolicyList `json:"deviceDiscoveryPolicies,omitempty"`
+}
+
+type AutoDetectVolumeStatus struct {
+	Status string `json:"status"`
+}
+
+type DeviceDiscoveryPolicyList struct {
+	Items []DeviceDiscoveryPolicy `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+type DeviceDiscoveryPolicy struct {
+	// Device type that should be used for auto detection. This would be one of the types supported
+	// by the local-storage operator. Initial possible types can be - crypt, raid1, raid4, raid5,
+	// raid10, multipath, disk, tape, printer, processor, worm, rom, scanner, mo-disk, changer,
+	// comm, raid, enclosure, rbc, osd, and no-lun.
+	DeviceType DeviceDiscoveryPolicyType `json:"deviceType"`
+	// File system type
+	// +optional
+	FSType string `json:"fsType,omitempty"`
+	// Volume mode. Raw or with file system
+	// + optional
+	VolumeMode PersistentVolumeMode `json:"volumeMode,omitempty"`
+	// DeviceInclusionSpec are the filtration rules for including a device in the device discovery
+	// +optional
+	DeviceInclusionSpec *DeviceInclusionSpec `json:"deviceInclusionSpec"`
+}
+
+// DeviceMechanicalProperty holds the device's mechanical spec. It can be rotational,nonRotational and
+// RotationalAndNonRotational
+type DeviceMechanicalProperty string
+
+const (
+	// The mechanical properties of the devices
+	Rotational                 DeviceMechanicalProperty = "rotational"
+	NonRotational              DeviceMechanicalProperty = "nonRotational"
+	RotationalAndNonRotational DeviceMechanicalProperty = "rotationalAndNonRotational"
+)
+
+type DeviceInclusionSpec struct {
+	// A regular expression that will be used to include certain devices
+	// For example - "^rbd[0-9]+p?[0-9]{0,}$"
+	// +optional
+	DeviceNamePattern string `json:"deviceNameFilter"`
+
+	// The devices of this mechanicalPropery will be included.
+	// By default it is RotationalAndNonRotational(or SSD/HDD both)
+	// +optional
+	DeviceMechanicalProperty DeviceMechanicalProperty `json:"deviceMechanicalProperty"`
+
+	// The minimum size of the device which needs to be included
+	// +optional
+	MinSize *int `json:"minSize"`
+
+	// The maximum size of the device which needs to be included
+	// +optional
+	MaxSize *int `json:"maxSize"`
+
+	// The list of device models which need to be included
+	// +optional
+	Models []string `json:"model"`
+
+	// Devices with below labels will be included
+	// +optional
+	Labels []string `json:"label"`
+}
+```
+
+An example of autoDetectVolume CR instance:
+```
+apiVersion: local.storage.openshift.io/v1
+kind: AutoDetectVolume
+metadata:
+  name: example-autodetect
+spec: 
+  storageClassName: example-storageclass
+  nodeSelector:
+  deviceDiscoverPolicies:
+    items:
+      - deviceType: part
+        fsType: ext4
+        volumeMode: RWO
```

Comment:
volumemode is typically raw block vs fs type.

Reply:
yup updated